core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -996,6 +1076,7 @@ private Releasable acquireLock(Term uid) {
     }
 
     private long loadCurrentVersionFromIndex(Term uid) throws IOException {
+        assert incrementIndexVersionLookup();
```

Comment:
we do there is a different one called `incrementVersionLookup` that is incremented in both situations